File: src/Domain/Entities/Activity.cs
using StartupConnect.Domain.Enums;

namespace StartupConnect.Domain.Entities;

public sealed class Activity : BaseEntity
{
    public Guid ProjectId { get; set; }

    public Project Project { get; set; } = null!;

    public Guid? ActorUserId { get; set; }

    public User? ActorUser { get; set; }

    public ActivityType Type { get; set; }

    public ActivityVisibility Visibility { get; set; } = ActivityVisibility.MembersOnly;

    public string Title { get; set; } = string.Empty;

    public string? Message { get; set; }

    public string? TargetType { get; set; }

    public Guid? TargetId { get; set; }

    public string? MetadataJson { get; set; }
}

File: frontend/src/types/activity.ts
export interface ActivityDto {
  id: string;
  projectId: string;
  projectTitle: string;
  actorUserId?: string;
  actorName?: string;
  type: string;
  visibility: string;
  title: string;
  message?: string;
  targetType?: string;
  targetId?: string;
  createdAt: string;
}

export interface ActivityListResponse {
  items: ActivityDto[];
  total: number;
  page: number;
  pageSize: number;
}
